src/screens/Platform/DAppDisclaimer.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import BottomModal from "../../components/BottomModal";
+import AppIcon from "./AppIcon";
+import LedgerIcon from "../../icons/Ledger";
+import {
+  Flex,
+  IconBox,
+  Text,
+  Checkbox,
+  Icons,
+  Button,
+} from "@ledgerhq/native-ui";
+
+export type Props = {
+  closeDisclaimer: () => void;
+  disableDisclaimer: () => void;
+  onContinue: () => void;
+  isOpened: boolean;
+  name: string | null;
+  icon?: string | null;
+};
+
+const IconsSeparator = React.memo(() => (
+  <Flex flexDirection="row" justifyContent="center" alignItems="center" mx={2}>
+    {Array(6)
+      .fill(undefined)
+      .map((_, i) => (
+        <Flex
+          key={i}
+          width="3px"
+          height={1}
+          marginX={2}
+          backgroundColor="neutral.c40"
+        />
+      ))}
+  </Flex>
+));
+
+const DAppDisclaimer = ({
+  closeDisclaimer,
+  isOpened,
+  disableDisclaimer,
+  onContinue: next,
+  icon,
+  name,
+}: Props) => {
+  const { t } = useTranslation();
+  const [disableDisclaimerChecked, setDisableDisclaimerChecked] = useState(
+    false,
+  );
+
+  const onClose = useCallback(() => {
+    closeDisclaimer();
+  }, [closeDisclaimer]);
+
+  const onContinue = useCallback(() => {
+    if (disableDisclaimerChecked) {
+      disableDisclaimer();
+    }
+    closeDisclaimer();
+    next();
+  }, [disableDisclaimerChecked, closeDisclaimer, disableDisclaimer, next]);
+
+  return (
+    <BottomModal isOpened={isOpened} onClose={onClose}>
+      <Flex
+        flexDirection="row"
+        justifyContent="center"
+        alignItems="center"
+        mb={10}
+      >
+        <IconBox iconSize={28} boxSize={40} Icon={LedgerIcon} />
+        {icon && (
+          <>
+            <IconsSeparator />
+            <Flex mx={2}>
+              <AppIcon size={40} icon={icon} />
+            </Flex>
+          </>
+        )}
+      </Flex>
+
+      <Text
```

Comment:
The title and description is already templated in the BottomModal component :
https://ledger-live-ui-native.vercel.app/?path=/story/layout-modal--bottomdrawer

Maybe there was a problem with it ?

Reply:
Yeah, not a big deal but the component created on top of the BottomDrawer component is more restrictive than the component we created for our design system. Typically Icon/Title... props aren't exposed.
I suggest merging this PR then I'll rework the component `BottomModal` component mobile-side to accept all the props defined by `BottomDrawer`

Reply:
In such cases, I just use BottomDrawer from ui directly, maybe it's not the right thing to do ? 😬

I agree ✅